tests/test_mask_parameter.py

```diff
@@ -0,0 +1,111 @@
+"""Various tests for the maskParameter"""
+
+import os
+import unittest
+import pytest
+from io import StringIO
+import astropy.time as time
+import astropy.units as u
+import numpy as np
+
+from pint.models.model_builder import get_model
+from pint.models.parameter import maskParameter
+from pint.toa import get_TOAs
+from pinttestdata import datadir
+
+import copy
+
+
+class TestParameters(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        os.chdir(datadir)
+        self.toas = get_TOAs("J1713+0747_NANOGrav_11yv0_short.tim")
+
+    def test_mjd_mask(self):
+        mp = maskParameter("test1", key="mjd", key_value=[54000, 54100])
+        assert mp.key == "mjd"
+        assert mp.key_value == [54000, 54100]
+        assert mp.value == None
+        select_toas = mp.select_toa_mask(self.toas)
+        assert np.all(self.toas.table["mjd_float"][select_toas] <= 54100)
```

Comment:
Can you check equality - that the set selected by `select_toa_mask` is the same as the set of TOAs with the right frequencies? You might find a utility function useful here, since that's what one wants to check for almost all these selections.

Reply:
Can you explain more about this? I am confused about the set of TOAs with the right frequencies.

Reply:
So `select_toa_mask` returns a list of TOA indices that match the mask. Right now you check that all the TOAs returned match the mask criterion - but that would pass if it returned no TOAs. Why not use `np.where` to get a list of TOAs that match the criterion, and compare it to the list returned by `select_toa_mask`? They should be exactly equal.

Reply:
The return of `select_toa_mask` returns the indices of selection, so I compare the indices between the result and `np.where()` return instead.